src/sdk/namespace_fs.js

```diff
@@ -299,6 +298,26 @@ class NamespaceFS {
      * @param {ListParams} params 
      */
     async list_objects(params, object_sdk) {
+        return this._list_objects(params, object_sdk, false);
+    }
+
+    async list_object_versions(params, object_sdk) {
+        return this._list_objects(params, object_sdk, true);
+    }
+
+    async _list_version_dir_objects(fs_context, results, entry_name, dir_key) {
+        const filePath = path.normalize(path.join(this.bucket_path, '/', dir_key, entry_name));
+        const versions = await nb_native().fs.readdir(fs_context, filePath);
+
+        versions.forEach(version => {
+            const v = {
+                key: path.normalize(path.join(dir_key, entry_name, '/', version.name)),
```

Comment:
do we need the full path for stat? 
I think also it could be added to the cache with a small change

Reply:
This method is not relevant since `process_dir` is used to update the version list.